anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+
+
+Arguments
+---------
+
+.. program:: anaDACScan.py
+
+.. option:: infilename
+
+    Name of the input file root.
+
+.. option:: --assignXErrors  
+
+    Use the dacValX_Err values stored in the input file
+
+.. option:: --calFileList
+  
+    Provide a file containing a list of calFiles in the following format. Example:: 
+
+    OH1 calFileOH1
+    OH2 calFileOH2
+    OH3 calFileOH3
+    etc
+
+.. option:: --outfilename
+
+    Name of the output root file. Default is DACFitData.root.
+
+Example
+-------
+
+.. code-block:: bash
+
+    anaDACScan.py /afs/cern.ch/user/d/dorney/public/v3Hack/dacScanV3.root  --calFileList calibration.txt --assignXErrors
+
+
+Environment
+-----------
+
+
+Internals
+---------
+
+
+"""
+
+if __name__ == '__main__':
+
+    import ROOT as r
+    
+    from gempython.gemplotting.utils.anautilities import parseCalFile
+    from gempython.utils.nesteddict import nesteddict
+    from gempython.gemplotting.utils.anautilities import make3x8Canvas
+    
+    from gempython.gemplotting.mapping.chamberInfo import chamber_config
+
+    import os
+    
+    import argparse
+    
+    parser = argparse.ArgumentParser(description='Arguments to supply to anaDACScan.py')
+
+    parser.add_argument('infilename', type=str, help="Filename from which input information is read", metavar='infilename')
+    parser.add_argument("--calFileList", type=str, help="File specifying which calFile to use for each OH. Format: OH1 filenameOH1 <newline> OH2 filenameOH2 <newline> etc", metavar="calFileList")
+    parser.add_argument('-o','--outfilename', dest='outfilename', type=str, default="DACFitData.root", help="Filename to which output information is written", metavar='outfilename')
+    parser.add_argument('--assignXErrors', dest='assignXErrors', action='store_true', help="Whether to assign errors for the X variable (which is actually plotted on the y-axis)")
+
+    args = parser.parse_args()
+
+    print("Analyzing: '%s'"%args.infilename)
+
+    # Set default histogram behavior
+    r.TH1.SetDefaultSumw2(False)
+    r.gROOT.SetBatch(True)
+    r.gStyle.SetOptStat(1111111)
+
+    dacScanFile = r.TFile(args.infilename)
+
+    import numpy as np
+    import root_numpy as rp
+    
+    list_bNames = ['link']
+    ohArray = rp.tree2array(tree=dacScanFile.dacScanTree, branches=list_bNames)
+    ohArray = np.unique(ohArray['link'])
+    
+    dataPath = os.getenv("DATA_PATH") 
+    
+    if len(args.infilename.split('/')) > 1 and len(args.infilename.split('/')[len(args.infilename.split('/')) - 2].split('.')) == 5:
+        scandate = args.infilename.split('/')[len(args.infilename.split('/')) - 2]
+    else:    
+        scandate = 'noscandate'
+
+    for oh in ohArray:
+        os.system("mkdir -p "+ dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate)
```

Comment:
if `scandate == `noscandate` then I wouldn't call this but instead write to `ELOG_PATH/chamber_config[oh]`

Reply:
```python
elogPath = os.getenv('ELOG_PATH')
```